src/cave-square.c

```diff
@@ -912,6 +912,18 @@ bool square_isbelievedwall(struct chunk *c, struct loc grid)
 	return !square_isprojectable(player->cave, grid);
 }
 
+
+/**
+ * Checks if a square is known by the player to be passable
+ */
+bool square_isknownpassable(struct chunk *c, struct loc grid)
```

Comment:
Sorry I didn't see this in the prior review, but the way this function is used in the navigation code, it should be equivalent to square_isknown(cave, grid) && square_ispassable(player->cave, grid).  With the way square_isknown() currently works (related issue https://github.com/angband/angband/issues/5879 ), I would change the return statement to use square_ispassable(player->cave, grid).